config/action.d/cloudflare.conf

```diff
@@ -60,7 +60,7 @@ actionban = curl -s -o /dev/null -X POST -H 'X-Auth-Email: <cfuser>' -H 'X-Auth-
 # API v4
 actionunban = curl -s -o /dev/null -X DELETE -H 'X-Auth-Email: <cfuser>' -H 'X-Auth-Key: <cftoken>' \
               https://api.cloudflare.com/client/v4/user/firewall/access_rules/rules/$(curl -s -X GET -H 'X-Auth-Email: <cfuser>' -H 'X-Auth-Key: <cftoken>' \
-              'https://api.cloudflare.com/client/v4/user/firewall/access_rules/rules?mode=block&configuration_target=ip&configuration_value=<ip>&page=1&per_page=1' | cut -d'"' -f6)
+              'https://api.cloudflare.com/client/v4/user/firewall/access_rules/rules?mode=block&configuration_target=ip&configuration_value=<ip>&page=1&per_page=1' | jq -r '.result[0].configuration.value')
```

Comment:
Is the value returned from json via `jq -r '.result[0].configuration.value'` not the same as `<ip>`?
Because if yes, I don't know why we need `jq` here...

Reply:
YEs it is.
But cut does not work anymore.

Reply:
What I don't understand, if you read the IP (with the second `$(curl... | jq...)`), then why simply not:
```diff
 actionunban = curl -s -o /dev/null -X DELETE -H 'X-Auth-Email: <cfuser>' -H 'X-Auth-Key: <cftoken>' \
-              https://api.cloudflare.com/client/v4/user/firewall/access_rules/rules/$(curl -s -X GET -H 'X-Auth-Email: <cfuser>' -H 'X-Auth-Key: <cftoken>' \
-              'https://api.cloudflare.com/client/v4/user/firewall/access_rules/rules?mode=block&configuration_target=ip&configuration_value=<ip>&page=1&per_page=1' | jq -r '.result[0].configuration.value')
+              https://api.cloudflare.com/client/v4/user/firewall/access_rules/rules/<ip>
```
Or (if I take a look to discussion, compared to solution of @wpkc) possibly should it be:
```diff
-... | jq -r '.result[0].configuration.value'
+... | jq -r '.result[0].id'
```
Then is it not the IP what API expects by DELETE, but some internal id of cloudflare?
Can you show me which value exactly needs this call?

Reply:
Yes, CloudFlare API v4 expects and access rule ID for the delete call.  The reason jq is not used is because you can rely on it being available. But you can always count on python being available if fail2ban is working. So...
 python -c "import sys, json; print json.load(sys.stdin)['result'][0]['id'];" 
...should always work.